chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -993,3 +1013,18 @@ func (b *BitcoindNotifier) GetBlock(hash *chainhash.Hash) (*wire.MsgBlock,
 
 	return b.blockCache.GetBlock(hash, b.chainConn.GetBlock)
 }
+
+// SubscribeMempoolSpent allows the caller to register a subscription to watch
+// for a spend of an outpoint in the mempool.The event will be dispatched once
+// the outpoint is spent in the mempool.
+//
+// NOTE: part of the MempoolWatcher interface.
+func (b *BitcoindNotifier) SubscribeMempoolSpent(
+	outpoint *wire.OutPoint) *chainntnfs.MempoolSpendEvent {
+
+	// TODO(yy): figure out how it interacts with `RegisterSpendNtfn`.
```

Comment:
I don't think we want to have this funnel into `RegisterSpendNtfn`: in the past we removed mempool based notifications as it makes it harder to play several transaction in flight. 

Instead, I think this should be a disticnt sub-system used as a look-side check for relevant sub-systems. In this case, we'd use it in the link when the outgoing link hits the chain.

Reply:
yeah added that comment because I wasn't sure about the implications. Previously the system behaves based on confirmed spends, and we now wanna change parts of it to behave based on unconfirmed spends, wasn't sure how we should change the behaviors now that some of the system states have already been changed based on unconfirmed spends. Anyway maybe the last commit has more insights.